libsolidity/ast/Types.cpp

```diff
@@ -2003,6 +2003,16 @@ vector<tuple<VariableDeclaration const*, u256, unsigned>> ContractType::stateVar
 	return variablesAndOffsets;
 }
 
+vector<VariableDeclaration const*> ContractType::immutableVariables() const
+{
+	vector<VariableDeclaration const*> variables;
+	for (ContractDefinition const* contract: boost::adaptors::reverse(m_contract.annotation().linearizedBaseContracts))
+		for (VariableDeclaration const* variable: contract->stateVariables())
```

Comment:
I don't think this is relevant here - the stateVariables function above also does not respect it.

Reply:
Well, that's the point, here you're also getting those. Is that wanted?

Reply:
Just because they are not in scope doesn't mean that the compiler should not consider them. For storage, for example, this is important to properly calculate the layout.

Reply:
Ok